src/DynamoCore/Graph/Nodes/CustomNodes/ICustomNodeManager.cs

```diff
@@ -9,6 +9,6 @@ public interface ICustomNodeManager
     {
         IEnumerable<CustomNodeInfo> AddUninitializedCustomNodesInPath(string path, bool isTestMode, bool isPackageMember = false);
         Guid GuidFromPath(string path);
-        bool TryGetFunctionWorkspace(Guid id, bool isTestMode, out ICustomNodeWorkspaceModel ws);
+        bool TryGetFunctionWorkspace(Guid id, bool isTestMode, out ICustomNodeWorkspaceModel ws, Engine.LibraryServices libraryServices = null);
```

Comment:
is library services actually optional?

Reply:
yes

Reply:
I think that Library Service can't be optional, because NodeReadConverter class needs non nullable LibraryServices at constructor.